### ISSUE TITLE
rule(MD077): nested ordered lists that don't start at `1.` cause false positives

### ISSUE THREAD

**Issue by eread** [rvben/rumdl#750]
### rumdl version

0.2.39

### Installation method

mise

### Bug description

Rule MD077 (https://github.com/rvben/rumdl/blob/db2c2048f4f6df06e6a38cf5c7896ef6d1e677df/docs/md077.md) checks on list continuation indentation.

However, if an ordered list is indented and doesn't start with `1.`, the rule seems to lose track of which list continuation text belongs to. It appears that `rumdl` looks for the last list that started with `1.` for the list it considers the list continuation text belongs to.

### Steps to reproduce

1. Create a file called `test.md` with this content:
   ```markdown
   ---
   title: A heading
   ---
   
   1. A level 1 list item.
      2. A level 2 list item.
      3. A level 2 list item.
         Some continuation text.
   ```
1. Create a `.rumdl.toml` file with this content:
   ```toml
   [global]
   enable = ["MD077"]
   ```
1. Run `rumdl check test.md`.

### Expected behavior

I'd expect no violations from that Markdown given that configuration.

### Actual behavior

`rumdl` returns:

```shell
test.md:8:1: [MD077] Continuation line over-indented (expected 3, found 6) [*]
```

### Configuration

```toml
[global]
enable = ["MD077"]
```

### Operating system

macOS

**Comment by rvben** [comment]
Thanks for the report, but I think rumdl is right here, and the reason is upstream of MD077.

In CommonMark an ordered list can only interrupt a paragraph if it starts with `1.`. In your example `2.` and `3.` are indented under `1.`, so they never begin a list. The whole block is a single `<li>` whose content is one paragraph, and `2.`/`3.` are literal text in that paragraph. I checked the token stream rather than eyeballing it: one list, one item, one paragraph, no nested list anywhere.

So there is no nested list for MD077 to lose track of. The content column of that single item is 3, the continuation lines sit at 6, and that is what gets flagged. I also verified the autofix is render-preserving: the HTML before and after is identical.

Your own second example is the fix. Starting the nested list at `1.` makes it a real nested list, and rumdl reports nothing on it.

**Comment by eread** [comment]
> Thanks for the report, but I think rumdl is right here, and the reason is upstream of MD077.
> 
> In CommonMark an ordered list can only interrupt a paragraph if it starts with `1.`. In your example `2.` and `3.` are indented under `1.`, so they never begin a list. The whole block is a single `<li>` whose content is one paragraph, and `2.`/`3.` are literal text in that paragraph. I checked the token stream rather than eyeballing it: one list, one item, one paragraph, no nested list anywhere.
> 
> So there is no nested list for MD077 to lose track of. The content column of that single item is 3, the continuation lines sit at 6, and that is what gets flagged. I also verified the autofix is render-preserving: the HTML before and after is identical.
> 
> Your own second example is the fix. Starting the nested list at `1.` makes it a real nested list, and rumdl reports nothing on it.

@rvben That sounds fine. Thanks for taking a look.
